multiversx_sdk/network_providers/account_awaiter.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import time
+from typing import Callable, Optional, Protocol, Union
+
+from multiversx_sdk.core.address import Address
+from multiversx_sdk.network_providers.constants import (
+    DEFAULT_ACCOUNT_AWAITING_PATIENCE_IN_MILLISECONDS,
+    DEFAULT_ACCOUNT_AWAITING_POLLING_TIMEOUT_IN_MILLISECONDS,
+    DEFAULT_ACCOUNT_AWAITING_TIMEOUT_IN_MILLISECONDS)
+from multiversx_sdk.network_providers.errors import \
+    ExpectedAccountConditionNotReachedError
+from multiversx_sdk.network_providers.resources import AccountOnNetwork
+
+ONE_SECOND_IN_MILLISECONDS = 1000
+
+logger = logging.getLogger("account_awaiter")
+
+
+class IAccountFetcher(Protocol):
+    def get_account(self, address: Address) -> AccountOnNetwork:
+        ...
+
+
+class AccountAwaiter:
```

Comment:
Will we integrate it in the network providers in a next PR?

Reply:
Forgot about it. Will do it in this PR. Thanks!